### ISSUE TITLE
Support Windows

### ISSUE THREAD

**Issue by ImSejin** [sp1thas/dropboxignore#21]
Does this support on Windows?

**Comment by sp1thas** [comment]
Hello @ImSejin 

Currently, dropboxignore is available only for macOS and Linux. Windows support is in my long-term plans for this project.

**Comment by codespearhead** [comment]
@sp1thas 

Does it have anything to do with the [rebuild of the module](https://github.com/sp1thas/dropboxignore/projects/1) in Python?

I don't quite get why it allegedly doesn't work on Windows machines.

**Comment by sp1thas** [comment]
@Paguiar735 exactly! This refactoring will make the `dropboxignore` a cross-platform tool.

The current implementation can not work on windows because it's implemented using shell and uses various libraries which are available on unix-like operating systems only.

**Comment by ankitp94** [comment]
I also vote for Windows support.

**Comment by codespearhead** [comment]
> I also vote for Windows support. 

I'm not the owner of the repo, nor am I a maintainer thereof.

However, given its excellent quality and effectiveness, I'm thinking of rewriting it in Go in another repo not only because it's way faster at scale, but also because Apple's recent introduction of the M1 processor might hinge the development of the software, as well as the release of potentially new features of said software to Apple users.

One thing I'm not so sure about is how we should go about the license in that other repo, since I'm only used to publishing under either APACHE-2.0 or MIT.

@sp1thas , can you enlighten us?

In either case, if any of you is willing to help, let me know by replying to this message

**Comment by sp1thas** [comment]
Hey @Paguiar735 Thanks for your feedback. It's not clear to me if your proposal is about refactoring the current project or creating a new one. If your are intrested on creating a new project similar to the current one, you are free to choose any licence you want. The initial thought about this project was to implement it in Shell in order to have the minimum dependencies. As time goes on, I relialize that windows support is really important, therefore, the refoctoring of this project using a cross-platform programming language should be planed asap. There is [already a idea](https://github.com/sp1thas/dropboxignore/projects) to refactor it using Python. Although I don't have previous experience in Go Lang, I will be more than happy to contribute in any way.

**Comment by sp1thas** [comment]
fyi: The refactoring of the project and the cross-platform support (including windows is in progress).

**Comment by codespearhead** [comment]
Since the time I gave that suggestion, I had the opportunity to work with both Go and Ruby, which made me realize neither is a good fit for the project: compiled [Go binaries are almost always wrongly flagged as virus](https://go.dev/doc/faq#virus) and ruby gems (project dependencies) need root access to be installed, unless you set up a virtual environment, which is also something that needs root access to be performed.

Hence, Python seems to be the most reasonable choice, given that it's more likely users already have it installed on their machine then a JRE, for instance, which might be an overkill.

I plan to work on that in late 2023.

Let us know in this thread if anyone else is willing to help. 🙂
